test/com/google/enterprise/adaptor/ConfigTest.java

```diff
@@ -330,4 +332,107 @@ public void testPropertiesParseEscapeASlash6() throws Exception {
     config.load(configFile);
     assertEquals("\\\t\\\f", config.getValue("slash"));
   }
+
+  @Test
+  public void testValidValuesForTimeouts() throws Exception {
+    configFile.setFileContents("gsa.hostname=not_used\n"
+                               + "adaptor.incrementalPollPeriodSecs=1700\n"
+                               + "adaptor.docHeaderTimeoutSecs=50\n"
+                               + "adaptor.docContentTimeoutSecs=190\n"
+                               + "adaptor.heartbeatTimeoutSecs=\n");
+    config.load(configFile);
+    assertEquals(TimeUnit.SECONDS.toMillis(1700),
+        config.getAdaptorIncrementalPollPeriodMillis());
+    assertEquals(TimeUnit.SECONDS.toMillis(50),
+        config.getAdaptorDocHeaderTimeoutMillis());
+    assertEquals(TimeUnit.SECONDS.toMillis(190),
+        config.getAdaptorDocContentTimeoutMillis());
+    assertEquals(TimeUnit.SECONDS.toMillis(50),
+        config.getAdaptorHeartbeatTimeoutMillis());
+  }
+
+  @Test
+  public void testPropertiesIncrementalPollPeriodSecs() throws Exception {
+    // incrementalPollPeriodSecs=0 and incrementalPollPeriodSecs=-15 are invalid
+    String putInConfig = " gsa.hostname=not_used\n"
+                         + "adaptor.incrementalPollPeriodSecs=";
+    String invalidValuesToVerify[] = {"0", "-15", "NotValidValue", "",
+                                      Long.toString(Long.MAX_VALUE + 1)};
```

Comment:
Long.MAX_VALUE + 1 = Long.MIN_VALUE
What you want is (Long.MAX_VALUE/1000)+1 , which will overflow when converted to milliseconds.